packages/noya-colorpicker/src/utils/colorAt.ts

```diff
@@ -0,0 +1,61 @@
+import { RgbaColor } from '../types';
+
+export interface PositionHsva {
+  color: RgbaColor;
+  pos: number;
+}
+
+export const RGBA_MAX = { r: 256, g: 256, b: 256, a: 1 };
+/**
+ * Linearly compute the step size between start and end (not normalized)
+ */
+ function stepize(start: RgbaColor, end: RgbaColor, steps: number) {
+    let step: RgbaColor = RGBA_MAX;
+
+    let key: keyof RgbaColor;
+    for (key in start) {
+      step[key] = steps === 0 ? 0 : (end[key] - start[key]) / steps;
+    }
+
+    return step;
+  }
+
+  /**
+   * Compute the final step color
+   */
+  function interpolate(
+    step: RgbaColor,
+    start: RgbaColor,
+    i: number,
+    max: RgbaColor,
+  ) {
+    let color: RgbaColor = RGBA_MAX;
+
+    let k: keyof RgbaColor;
+    for (k in start) {
+        color[k] = step[k] * i + start[k];
+        color[k] = color[k] < 0 ? color[k] + max[k] : color[k];
+    }
+
+    return color;
+  }
+
+  export function colorAt(stops: PositionHsva[], pos: number, max: RgbaColor) {
```

Comment:
I want to play around with this a bit you rebase, cause I added an interpolate util a couple days ago that we might be able to use to simplify this